backend/tests/services/GenerateReporter/GenerateReportService.test.ts

```diff
@@ -0,0 +1,806 @@
+    },
+    {
+      reasonName: "sit_env_down",
+      percentage: 0,
+    },
+    {
+      reasonName: "priority_change",
+      percentage: 0,
+    },
+    {
+      reasonName: "solution_review",
+      percentage: 0,
+    },
+    {
+      reasonName: "pr_review",
+      percentage: 0,
+    },
+    {
+      reasonName: "question_to_be_answered",
+      percentage: 0,
+    },
+    {
+      reasonName: "take_leave",
+      percentage: 0,
+    },
+    {
+      reasonName: "others",
+      percentage: 0.16,
+    },
+  ],
+};
+const response: GenerateReporterResponse = {
+  velocity: {
+    velocityForSP: "6",
+    velocityForCards: "3",
+  },
+  cycleTime: {
+    averageCircleTimePerCard: "3.93",
+    averageCycleTimePerSP: "1.97",
+    totalTimeForCards: 11.8,
+    swimlaneList: [
+      {
+        optionalItemName: "To do",
+        averageTimeForSP: "0.33",
+        averageTimeForCards: "0.65",
+        totalTime: "1.95",
+      },
+      {
+        optionalItemName: "In Dev",
+        averageTimeForSP: "1.11",
+        averageTimeForCards: "2.23",
+        totalTime: "6.68",
+      },
+      {
+        optionalItemName: "Testing",
+        averageTimeForSP: "0.54",
+        averageTimeForCards: "1.07",
+        totalTime: "3.22",
+      },
+      {
+        optionalItemName: "Done",
+        averageTimeForSP: "4.80",
+        averageTimeForCards: "9.60",
+        totalTime: "28.80",
+      },
+      {
+        optionalItemName: "Block",
+        averageTimeForSP: "0.32",
+        averageTimeForCards: "0.63",
+        totalTime: "1.90",
+      },
+    ],
+  },
+  changeFailureRate: undefined,
+  classification: [],
+  completedCardsNumber: completedCardsNumber,
+  standardDeviation: standardDeviation,
+  blockedAndDevelopingPercentage: blockedAndDevelopingPercentage,
+  deploymentFrequency: undefined,
+  leadTimeForChanges: undefined,
+  meanTimeToRecovery: undefined,
+  latestSprintBlockReason: latestSprintBlockReason,
+};
+const statistics = new SprintStatistics(
+  completedCardsNumber,
+  standardDeviation,
+  blockedAndDevelopingPercentage,
+  latestSprintBlockReason
+);
+
+const pipeLineMetrics = [
+  RequireDataEnum.CHANGE_FAILURE_RATE,
+  RequireDataEnum.DEPLOYMENT_FREQUENCY,
+  RequireDataEnum.MEAN_TIME_TO_RECOVERY,
+].map((metric) => metric.toLowerCase());
+const codebaseMetrics = [RequireDataEnum.LEAD_TIME_OF_CHANGES].map((metric) =>
+  metric.toLowerCase()
+);
+const kanbanMetrics = [
+  RequireDataEnum.VELOCITY,
+  RequireDataEnum.CYCLE_TIME,
+  RequireDataEnum.CLASSIFICATION,
+].map((metric) => metric.toLowerCase());
+
+describe("GenerateReportService", () => {
+  afterEach(() => sinon.restore());
+  const service = new GenerateReportService();
+  const serviceProto = Object.getPrototypeOf(service);
+  serviceProto.kanbanMetrics = kanbanMetrics;
+  serviceProto.pipeLineMetrics = pipeLineMetrics;
+  serviceProto.codebaseMetrics = codebaseMetrics;
+  it("generate report with velocity & cycle time & classification", async () => {
```

Comment:
the test description should be more clear, maybe you can use style like "should xxx when xxx"